ebu_tt_live/bindings/_ebuttdt.py

```diff
@@ -319,11 +319,12 @@ def as_timedelta(cls, instance):
         :param instance:
         :return:
         """
-        hours, minutes, seconds, milliseconds = map(
-            lambda x: cls._int_or_none(x),
-            cls._groups_regex.match(instance).groups()
-        )
-        return timedelta(hours=hours, minutes=minutes, seconds=seconds, milliseconds=milliseconds)
+        hours_str, minutes_str, seconds_str, seconds_fraction_str = [x for x in cls._groups_regex.match(instance).groups()]
+        milliseconds = seconds_fraction_str and cls._int_or_none('{:0<3}'.format(seconds_fraction_str)[:3]) or 0
+        return timedelta(hours=cls._int_or_none(hours_str), 
```

Comment:
Hmm, is there a specific reason why `_int_or_none` is called here (and at other locations)? The regex should already ensure that the affected value is a valid integer, shouldn't it?

Reply:
No, it doesn't. If the group is missing then it returns None and then we get an exception downstream. In particular the last group can be missing.

Reply:
If the regex doesn't match, the `match` call [already returns `None` instead of a MatchObject](https://docs.python.org/2.7/library/re.html#re.RegexObject.match) (which would provide the `groups` method). So an exception is thrown in line 322 then (tested with value `10:00X:00.00`):

    AttributeError: 'NoneType' object has no attribute 'groups'

Consequently the invokations of `_int_or_none` have no effect here and could be removed.

Reply:
(except in case of the last group I think)

Reply:
Did you test with `10:00:00`?

Reply:
With `10:00:00` it indeed works with `_int_or_none` replaced by `int`. Which makes sense, as `seconds_fraction_str` is `None`, setting `milliseconds` explicitely to zero in line 323.

[BTW I always get a bit confused by this "C and x or y" statements without brackets and have to look up Python's operator precedence. E.g. "x if C else y" would be faster to perceive and could be a better option for future changes.]

Reply:
`_int_or_none` removed in 50ce306.